openstack/datadog_checks/openstack/openstack.py

```diff
@@ -689,19 +689,29 @@ def get_stats_for_single_project(self, project):
         def _is_valid_metric(label):
             return label in PROJECT_METRICS
 
+        project_name = None
+        if 'name' in project:
+            project_name = project['name']
+
+        self.log.debug("Collecting metrics for project. name: {0} id: {1}".format(project_name, project['id']))
+
         url = '{0}/limits'.format(self.get_nova_endpoint())
         headers = {'X-Auth-Token': self.get_auth_token()}
         server_stats = self._make_request_with_auth_fallback(url, headers, params={"tenant_id": project['id']})
 
         tags = ['tenant_id:{0}'.format(project['id'])]
-        if 'name' in project:
+
+        if project_name:
             tags.append('project_name:{0}'.format(project['name']))
 
         for st in server_stats['limits']['absolute']:
             if _is_valid_metric(st):
                 metric_key = PROJECT_METRICS[st]
                 self.gauge("openstack.nova.limits.{0}".format(metric_key), server_stats['limits']['absolute'][st], tags=tags)
 
+    def get_stats_for_all_projects(self, projects):
+        for project in projects:
```

Comment:
I think we should do a sanity check if `projects` can be actually iterated, see below